src/react/utils.js

```diff
@@ -270,3 +276,46 @@ export function normalizeFunctionalComponentParamaters(func: ECMAScriptSourceFun
     }
   });
 }
+
+export function createReactHint(object: ObjectValue, propertyName: string, args: Array<Value>): ReactHint {
+  return {
+    object,
+    propertyName,
+    args,
+  };
+}
+
+export function getComponentTypeFromRootValue(realm: Realm, value: Value): ECMAScriptSourceFunctionValue {
+  let _valueIsKnownReactAbstraction = valueIsKnownReactAbstraction(value);
+  invariant(
+    value instanceof ECMAScriptSourceFunctionValue || _valueIsKnownReactAbstraction,
+    "only ECMAScriptSourceFunctionValue function values or known React abstract values are supported as React root components"
```

Comment:
This could possibly also be a compiler diagnostic if this invariant is being used to validate user input to `__registerReactComponentRoot`

Reply:
If I change the above comment you made to a CompilerDiagnostic, it should cover the "user" side of this problem as this utility function is only internal.